File: python/resistor-color-duo/resistor_color_duo.py
"""
Calculate the values for the first two colors on a resistor.
"""

color_bands = {
    'black': 0,
    'brown': 1,
    'red': 2,
    'orange': 3,
    'yellow': 4,
    'green': 5,
    'blue': 6,
    'violet': 7,
    'grey': 8,
    'white': 9,
}

def adjust_color(color):
    return color.lower().strip()

def value(colors):
    """
    Return the numeric value for the first two colors in the given colors list
    If there are less than two colors only the available colors will be counted. With zero for an empty list
    colors: a list of color names
    """
    ret = 0
    for index in range(2):
        if len(colors) >= (index + 1):
            ret = ret * 10 + color_bands[adjust_color(colors[index])]    
    return ret

File: go/resistor-color-duo/resistor_color_duo.go
// Decode the resistance of two resistor color bands
package resistorcolorduo

var colorBands [10]string = [10]string{"black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey", "white"}

// Return the resistance value of a resistor encoded by up to two color bands
// @param colors: Slice of string with color values to use.
// @returns: The calculated resistance value of up to the first two bands
func Value(colors []string) int {
	var result int = 0
	for index := 0; index < min(2, len(colors)); index++ {
		result = (result * 10) + ColorToValue(colors[index])
	}
	return result
}

// Find a color in the color bands array and return its index.
// @param color: the color to search for.
// @returns: -1 if not found otherwise the index of the desired color.
func ColorToValue(color string) int {
	for index := 0; index < len(colorBands); index++ {
		if colorBands[index] == color {
			return index
		}
	}
	return -1
}
